geopy/geocoders/base.py

```diff
@@ -185,6 +187,10 @@ def __init__(
                             else options.default_ssl_context)
 
         if self.proxies:
+            if isinstance(self.proxies, str):
+                self.scheme = self.proxies.split('://')[0]
+                self.proxies = {self.scheme: self.proxies}
```

Comment:
Given that we verify above that `self.scheme` is either `http` or `https`, I would suggest to specify proxy for both schemes instead of the current one.

Consider this situation:
```
g = SomeGeocoder(proxies='http://my_proxy_server:8080', scheme='https')
g.scheme = 'http'
```

After assignment proxy wouldn't be used, which is unexpected.

Reply:
I guess I wasn't clear enough on this one.

This is what I meant:
```
self.proxies = {'http': self.proxies, 'https': self.proxies}
```

That way my snippet above will work correctly.